### PR TITLE
feat: expose RichTextState selection and introduce Chat Input UI sample

### PR DESCRIPTION
**Pull request by mkeeda**
## Overview
This PR exposes the `selection` property from `RichTextState` and adds a comprehensive Chat Input UI sample that demonstrates applying all built-in formatting styles.

## Implementation Details

**richtext-editor**:
- Exposed `selection: TextRange` property in `RichTextState`.
- Added the `RichStringBuffer.editAttributes(selection: TextRange)` extension function, which safely handles right-to-left text selections by converting the range to `min until max`.
- Added integration tests in `RichTextEditorTest` to verify that formatting is correctly applied to reversed text selections.

**sample-app**:
- Introduced `ChatInputSample`, a multi-line `RichTextEditor` coupled with a horizontally scrollable formatting toolbar.
- The toolbar includes buttons for all currently supported built-in styles: Bold, Italic, Underline, Strikethrough, Text Color, Background Color, Font Size, Heading, Text Alignment, Blockquote, and Clear Formatting.
- Enhanced the UI layout to properly adjust when the software keyboard is open (using `adjustResize` and `imePadding()`), and positioned the toolbar below the text field to prevent overlap with OS context menus.
- Added UI tests (`ChatInputSampleTest`) using Robolectric to verify the initial layout state and button enablement based on text selection.

### CODE REVIEW

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt [hunk 1]
```diff
@@ -26,6 +26,13 @@ public class RichTextState(initialText: RichString) {
...
             )
 
+    /**
+     * The current selection range within the text field.
+     * Returns [androidx.compose.ui.text.TextRange.Zero] when no selection is active (cursor at position 0).
+     */
+    public val selection: androidx.compose.ui.text.TextRange
+        get() = textFieldState.selection
+
     public fun edit(block: RichStringBuffer.() -> Unit) {
```
Comment (mkeeda) [inline richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt:41]: No KDoc. Please write.

File: .idea/deploymentTargetSelector.xml [hunk 2]
```diff
@@ -4,6 +4,13 @@
...
         <option name="selectionMode" value="DROPDOWN" />
+        <DropdownSelection timestamp="2026-04-24T06:20:45.398492Z">
+          <Target type="DEFAULT_BOOT">
+            <handle>
+              <DeviceId pluginId="PhysicalDevice" identifier="serial=49061FDKD000PK" />
+            </handle>
+          </Target>
+        </DropdownSelection>
         <DialogSelection />
       </SelectionState>
```
Comment (mkeeda) [inline .idea/deploymentTargetSelector.xml]: This file contains a device-specific serial number (`serial=49061FDKD000PK`) and should not be part of this PR.

Consider adding `.idea/deploymentTargetSelector.xml` to `.gitignore` to prevent it from being tracked in the future, and revert this change from the PR.

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichStringBufferComposeExt.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Applies a set of attribute mutations to the range specified by the given [TextRange].
+ * This is a convenience overload that converts [TextRange] to [IntRange].
+ */
+public fun RichStringBuffer.editAttributes(
+    selection: TextRange,
+    editAction: AttributeEditScope.() -> Unit,
+) {
+    editAttributes(range = selection.min until selection.max, editAction = editAction)
+}
```
Comment (mkeeda) [inline richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichStringBufferComposeExt.kt:19]: The KDoc mentions converting to `min until max`, which is great. It might be worth also noting the behavior when the selection is collapsed (i.e., `min == max`), as this results in an empty range and effectively becomes a no-op. Documenting this explicitly helps callers understand whether they need to guard against collapsed selections themselves.

```kotlin
/**
 * Applies a set of attribute mutations to the range specified by the given [TextRange].
 * This is a convenience overload that converts [TextRange] to [IntRange].
 *
 * If the selection is collapsed (cursor with no selection), the resulting range is empty
 * and no attributes are modified.
 */
```

File: sample-app/src/main/java/dev/mkeeda/arranger/sampleApp/ChatInputSample.kt [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    hasSelection: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .horizontalScroll(rememberScrollState())
+                .padding(horizontal = 4.dp, vertical = 2.dp),
+    ) {
```
Comment (mkeeda) [inline sample-app/src/main/java/dev/mkeeda/arranger/sampleApp/ChatInputSample.kt:132]: Suggestion: The 11 `IconButton` blocks follow a nearly identical pattern (onClick → `state.edit { editAttributes(...) { action() } }`, icon resource, content description, enabled state). Extracting this into a helper data class and a loop could reduce boilerplate significantly:

```kotlin
private data class FormatAction(
    @DrawableRes val iconRes: Int,
    val contentDescription: String,
    val action: AttributeEditScope.() -> Unit,
)

// then iterate with formatActions.forEach { ... }
```

That said, this is a sample app and the current explicit approach is arguably more readable for documentation purposes, so this is purely optional.

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt [hunk 5]
```diff
@@ -26,6 +26,13 @@ public class RichTextState(initialText: RichString) {
...
         spans = richString.edit(block).spans
     }
```
Comment (mkeeda) [inline richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt:43]: Nit: `TextRange` is used as a fully qualified name here (`androidx.compose.ui.text.TextRange`), but it could be imported at the top of the file for consistency and conciseness:

```kotlin
import androidx.compose.ui.text.TextRange

public val selection: TextRange
    get() = textFieldState.selection
```

**Review by mkeeda — COMMENTED** [review]
(AI review)

## Summary

Good work on exposing the `selection` property and building a comprehensive Chat Input sample that showcases all built-in formatting styles. The `RichStringBuffer.editAttributes(TextRange)` extension is a clean convenience API, and the tests cover the important edge case of reversed selections well. CI is green ✅.

## Key Points

1. **`.idea/deploymentTargetSelector.xml`** — Contains a device-specific serial number and should be removed from this PR. Recommend adding it to `.gitignore`.
2. **FQN in `RichTextState.selection`** — Minor: `TextRange` is fully qualified inline; importing it at the top would be more concise.
3. **`editAttributes` KDoc** — Consider documenting the collapsed-selection (empty range) behavior explicitly.
4. **Toolbar boilerplate** — Optional suggestion to extract repeated `IconButton` patterns into a data-driven loop for maintainability (though the explicit style is fine for a sample app).
